### DISCUSSION TITLE
Cultivos frutales - orden de capas

### DISCUSSION THREAD

**Discussion by manudona92** [manudona92/MapantES#18] [General]
En algunas tiles de LZ estoy viendo que la capa de vegetación del KP está por encima del amarillo 100% de cultivo frutal (además de cultivo normal). Desde mi punto de vista no queda muy allá (se mezcla verde de KP con verde del símbolo de puntos), no muy legible. La vegetación de KP encima del amarillo 100% para sólo cultivo queda bien de momento. 

Voy a probar con renderizer a ver si ese amarillo 100% del simbolo de cultivo frutal se puede subir un poco, por encima del de la vegetación de KP. (no sé si esto lo pudiera hacer con renderizer, duplicando símbolo en OOM)
![image](https://user-images.githubusercontent.com/76484741/103466051-b59b5100-4d41-11eb-8e8a-4bfc5e20cce2.png)
![image](https://user-images.githubusercontent.com/76484741/103466058-c21fa980-4d41-11eb-91a3-894eff03bcf2.png)

**Comment by agustincaballero** [comment]
Esto es la última prueba que hice y lo dejé así, pero naturalmente no es definitivo.
Sobre cultivos es que tenemos que pensarlo bien, hay bastante casuística:
· Cultivos que no están en OSM
· Cultivos que están como farmland o vineyard (se supone que no son arboledas), pero que son arboledas
· Cultivos que están como orchard (arboledas) pero están como cultivos

**Reply by manudona92** [reply]
Creo que al final tenemos 3 opciones, de arriba a abajo según mi opinión: 

1. Quedarnos con lo que está. Cultivos (farmland o vineyard) con el amarillo 100% y frutales (orchard) con amarillo 100% con puntos verdes. El primero con la vegetación de KP por arriba y el segundo con la vegetación de KP por abajo (si se pudiera) 
2. Omitir la capa de cultivos del OSM. Al final los cultivos lo representaría la vegetación del KP, incluso con más precisión (los plataneros en la Gomera son verde 3, los viñedos en Oia son verde 1). También marcaría los límites de vegetación, por ejemplo un cultivo entre bosques, con más precisión. 
3.  Buscar otra solución. Pero no sé cuánto tiempo nos llevará ni si la podremos encontrar (SIOSE dscartado, BTN también. BTA sería entrar en poca generalización y trabajo de investigación). 

Estoy seguro que se os ocurren otras ideas, pero vuestras opiniones importan también. Con lo que he visto de Gomera y Lanzarote, y la omisión/poca precisión de OSM, me da igual si con la 1 o con la 2. Lógicamente, veo la opción 1 como algo muy simplificado ,desde un punto de vista orientador, donde me doy cuenta de los errores mientras la curva de nivel y la vegetación esté bien, y quizás estéticamente le da color al mapa. Si miras Rogaines o MTBO, el tema de la imprecisión de los cultivos afectaría más.

**Comment by agustincaballero** [comment]
He hecho una prueba con la opción 2, a mí en principio es la que más me convence. Es del noroeste de la cuadrícula 2703110, en La Gomera. Ahí hay zonas marcadas como farmland que en realidad son plataneras.

![image](https://user-images.githubusercontent.com/18677366/103555236-3bd5a580-4eb0-11eb-9ba3-252653e60b93.png)

Es decir, como has dicho en la opción 2, sin los puntos verdes y dejar que sea la vegetación KP la que marque que ahí hay árboles o no. Dejando el límite de cultivo, para que donde no haya vegetación se vea el "corte".

Pienso que en zonas de arboledas menos densas quedaría aún mejor ya que probablemente se vean separados y alineados, lo que sugerirá intuitivamente un cultivo de árboles.

**Reply by manudona92** [reply]
OK a todo. 

Respecto a a frutales o arboledas menos densas, sí, dependerá de la distancia entre árbol y la altura. Almendros y olivas que generalmente estan distantes, se verán como blanco y quizás pixelado la corona del árbol, pero quizás se vea como una sábana blanca si los árboles están a una cierta distancia. Naranjos, limoneros u otros cítricos, también la vid, se verá como verde 1-2 seguramente. sin verse árbol a árbol, ya que suelen estar más pegados.

**Comment by agustincaballero** [comment]
Reabro el tema de los cultivos porque he hecho una prueba con los datos de SIGPAC. 
Ésta es la ortofoto:
![image](https://user-images.githubusercontent.com/18677366/104133521-68902e00-5384-11eb-8af8-bca76fb7261e.png)

Esto es sin SIGPAC:
![image](https://user-images.githubusercontent.com/18677366/104133540-8067b200-5384-11eb-8adf-dbc9896c3f13.png)

Y esto es con SIGPAC:
![image](https://user-images.githubusercontent.com/18677366/104133555-96757280-5384-11eb-8a74-6e2caea53aff.png)

Tengo que ver cómo automatizar la descarga; las comunidades autónomas suelen tener la información para descargar, pero quiero ver si se puede descargar en tiempo real para no tener que almacenarla en nuestro servidor. Seguiré investigando.

**Comment by agustincaballero** [comment]
Ampliación: si queréis podéis evaluar alguna zona en el visor: http://sigpac.mapama.gob.es/fega/visor/

Hay que activar la capa Recintos (botón arriba a la derecha), y luego con el botón Consultas/Recintos, pinchando en un recinto da la información; como veréis, da información de tipo de cultivo. Lo bueno es que esta información está bastante actualizada porque de ella depende temas legales y de subvenciones. Por lo que he visto yo en zonas que conozco, es bastante completa.

**Comment by agustincaballero** [comment]
Ayer dejé bastante terminada la automatización de renderizar sembrados desde el SIGPAC. Esta noche pondré a re-renderizar todo para que veamos cómo queda en general.
Pero hay una cosa que creo que deberíamos pensar cómo mejorar: cuando las arboledas son densas (frutales, olivos), es bastante probable que aparezca la vegetación como todo blanco/verde. En estos casos, se hace difícil distinguir si eso es bosque o cultivo. Un ejemplo del Coto Maravillas:

![image](https://user-images.githubusercontent.com/18677366/104557894-43c8de80-5642-11eb-82b7-431a4e5c954d.png)

Lo que hay al sur de la rambla es un cultivo de frutales, pero en realidad da toda la impresión de ser bosque. 

Formas que se me ocurren de mejorar esto (pueden ser complementarias entre sí):
· Cambiar línea de límite de cultivo: podemos hacer lo que queramos:
- Ensancharla (pero si la dejamos negra puede crear confusión ya que suelen ir paralelas a los caminos
- Cambiarle el color (¿fuxia?)
- Cambiarle el trazo (¿puntos tipo cambio de vegetación?)

· Rellenar el área con algún patrón significativo. Algo tal que así:
![image](https://user-images.githubusercontent.com/18677366/104558900-dfa71a00-5643-11eb-9530-07fec1700035.png)

En este ejemplo he usado el fuxia (no me disgusta, evoca zona prohibida), pero podemos poner el color que queramos, y/o cambiar el patrón, poner puntos más gordos, poner otro símbolo en lugar de puntos, ponerlos a más distancia, etc.

**Reply by agustincaballero** [reply]
Éste es un ejemplo con una cuadrícula completa:
![vegetation mapant](https://user-images.githubusercontent.com/18677366/104559588-e5512f80-5644-11eb-9ed6-fe1db36e9ca4.png)

**Reply by manudona92** [reply]
La precisión geométrica de los cultivos es bastante buena, al menos en Murcia (!!)

Es posible renderizar este símbolo y poner este amarillo 100%+verde 3 (puntos) por encima de la vegetación del KP  ?? Solo seria para Orchard (frutales)
![image](https://user-images.githubusercontent.com/76484741/104627278-1ce8c780-5697-11eb-9488-df499b5fa402.png)

**Reply by agustincaballero** [reply]
Sí, la precisión del SIGPAC parece bastante buena. Un truco para revisar el SIGPAC es en la web http://sigpac.mapama.gob.es/fega/visor/ pulsar la tecla R, y al pasar el ratón por un recinto, nos da una referencia y al final dos letras:
![image](https://user-images.githubusercontent.com/18677366/104633462-77862180-569f-11eb-960e-867157e54671.png)

El significado de esas dos letras es el siguiente:

AG	Corrientes y Superficies de Agua
CA	Viales
CF	Asociación Cítricos-Frutales
CI	Citricos
CS	Asociación Cítricos-Frutales de Cáscara
CV	Asociación Cítricos-Viñedo
ED	Edificaciones
EP	Elemento del paisaje
FF	Asociación Frutales-Frutales De Cáscara
FL	Frutos Secos y Olivar
FO	Forestal
FS	Frutos Secos
FV	Frutos Secos y Viñedo
FY	Frutales
IM	Improductivos
IV	Invernaderos y cultivos bajo plastico
OC	Asociación Olivar-Cítricos
OF	Olivar - Frutal
OV	Olivar
PA	Pasto con Arbolado
PR	Pasto Arbustivo
PS	Pastizal
TA	Tierras Arables
TH	Huerta
VF	Viñedo - Frutal
VI	Viñedo
VO	Viñedo - Olivar
ZC	Zona Concentrada no incluida en la Ortofoto
ZU	Zona Urbana
ZV	Zona Censurada

Revisando por ejemplo la zona de Oia veo que incluso los viñedos parece que están bien catalogados. Yo iría a hacer una prueba de máximos, renderizando también los viñedos con su símbolo 414.0 (también por encima de la vegetación), y el resto con el símbolo 412.0.

Se puede re-renderizar todo así y vemos lo que sale, y a partir de ahí podemos quitar lo que se estime.

**Reply by manudona92** [reply]
Sí, el tema de los viñedos sería conveniente dejarlo en su símbolo. 

TA en el ejemplo de Murcia, están renderizados, yo lo veo bien. Creo que ganamos más que perdemos con este campo. En Alicante he observado y algunas zonas son tierras abandonadas por la desertificación, pero creo que son ejemplos aislados (??). He estado mirando en Valladolid, León que es cereal todavía activo y casi todo es TA, por lo que omitirlo quitaría todas estas tierras de Castilla (por ejemplo) del amarillo 100%. Son las zonas que he mirado, pero quizás en otras CCAA cambie. 

Respecto al borde, quizá haya que discutirlo, no afecta tanto pero si tiene errores de geometría. Queda obviamente mejor con el borde negro pero hay mucha parcela adyacente/paralela en el norte de España por ejemplo. Podemos ver con el ejemplo de Oia

Otra duda que tengo es el PS, hasta que punto lo cogemos o no. He visto rápidamente en zonas de Galicia y Pais Vasco y abunda mucho, es un amarillo 100% , lo que pasa que a veces se mete mucho en zonas más "boscosas".

**Reply by manudona92** [reply]
Esto es un ejemplo muy básico de Galicia, Todas esas miniparcelas son TA, incluso la que señalo en rojo. Lógicamente la vegetación del KP pasaría por encima y omitiria el error de clasificación (ya que es bosque esa en su mayoría). Aún así, con borde negro, yo veria demasiada información de división de parcelas, que quizas a 1:10K es innecesaria...

![image](https://user-images.githubusercontent.com/76484741/104649575-36007100-56b5-11eb-97bd-c29152c8bd92.png)

Aún así, como he dicho, con el nuevo renderizado de Oia podemos sacar más conclusiones..

**Reply by jarufe** [reply]
En el último ejemplo se puede ver la problemática de Galicia, sobre todo si van a quedar líneas negras dividiendo parcelas. La parcela a la izquierda del extremo de la primera flecha tiene una línea eléctrica. Las 2 siguientes a la izquierda son de terreno basto que estoy seguro que a la vista no ofrecen ninguna división. La siguiente parece ser un prado que sí tiene una división perceptible con el basto. La siguiente a la izquierda es bosque. Dependiendo de si el prado está encerrado en una valla o no, la línea de parcela dejaría esa limitación estupendamente representada. Luego sigue el mismo bosque pero con más divisiones, una de ellas dada por un camino.

Una cosa muy buena de la implementación de Agustín es poder re-renderizar sin tener que hacer todo el esfuerzo de procesamiento completo, por lo que estaría bien hacer distintas pruebas.

A mi, a priori, me parece que poner más información proveniente del SIGPAC (en este caso) es recargar demasiado el mapa con riesgo de ocultación de elementos o de cometer errores. Pero todo depende del objetivo final del que quiera usar el mapa. Si es para que un cartógrafo haga un mapa, yo no le daría más información. Si es para un excursionista, cuanta más información seguro que mejor. Si es para alguien que quiere ponerse un trazado y entrenar, fifty-fifty.

**Comment by agustincaballero** [comment]
He renderizado la zona del Coto Maravillas con los nuevos criterios.
También, en Oia, las cuatro cuadrículas más al sur, donde hay viñedos.
Yo creo que el resultado es bastante bueno, con algunas cosas a retocar; si queréis, id echando un vistazo, que ahora no puedo extenderme, esta tarde os comento algunas cosillas que he visto yo.

**Reply by manudona92** [reply]
Mmmm este tema es más complicado de lo que parecía :) 

A primera vista, de lo que veo es que el borde negro se mete mucho entre el oliva del privado urbano. En Galicia, se satura de bordes negros por todo el tema de minifundio. El punteado negro quizá sature de negro demasiado el mapa (?). Al menos en Murcia la calidad del delimitado es buena a salvo de un cultivo frutal que coge medio monte en las cuadriculas del este. Bueno, como esto puede ser un debate bueno, y a falta de otros detalles pequeños por definir, ¿Os parece si pongo la reunión con "consejo de sabios" la semana del 25 al 28 de enero? Lo ponemos más adelante por si acaso se nos come el tiempo eh, no habría problema. No sé por donde tienes pensado encaminar los nuevos test Agustín, quizás poniendo todo sin bordes? Es una pena porque el borde ayuda mucho a reconocer el límite, no solo de cultivo a amarillo 50% sino también de vid a frutal por ejemplo. No sé si en otros mapant hicieron algún tipo de "dissolve" (disolver polígono del SHP de mismo campo) previamente, para borrar todos los límites de minifundios adjuntos. El dissolve es una herramienta muy básica de GIS, pero no sé si lo podriamos ejecutar en servidor y tenerlo todo preparado disuelto antes de enviar a cliente. Bueno lo dejo ahí...como posible mejora, si es que pudieramos por capacidad.

**Comment by agustincaballero** [comment]
· Tenéis razón en que en casos como el de Galicia pues se ve mucha línea. La forma de hacer lo del "dissolve" (si te he entendido bien) es hacerlo por parcela en lugar de recinto, pero el problema es que en Galicia la mayoría de esos minifundios son parcelas distintas. Porque ir a un análisis GIS para unir polígonos contiguos, aunque factible, nos mete en una complejidad excesiva, creo.
· Se puede quitar los límites, sin problema, haré esa prueba renderizando varias zonas. Yo supongo que si dejamos que cada cultivo tenga su trama (puntos/rayas), se distinguirá suficientemente bien. Pero haré una prueba y lo vemos.
· Sobre "consejo de sabios", antes me aseguraría de que vamos a tener para enseñarles al menos algún trozo procesado ya con los 3 niveles de verde.

**Comment by manudona92** [comment]
Seguramente no habréis pensado en ella por la imposibilidad, pero solo suelto la piedra...
Meter una capa "cultivos" como la idea de los noruegos de "osm path" ? Claro, debe de estar en el orden bueno de colores (la curva por encima, etc) por eso creo que es imposible...pero mejor soltarlo. Mostrariamos todo sin cultivo pero a un cierto zoom se veria la capa cultivo, y el usuario la activa y la desactiva según su interes. 
![image](https://user-images.githubusercontent.com/76484741/105397242-1a4f1a80-5c21-11eb-982c-b3500cc89acc.png)

Por cierto, he visto que mapant.es ya está abierta :)
